File: framework/security-core/src/main/java/cn/bestwu/simpleframework/security/exception/IllegalUserException.java
package cn.bestwu.simpleframework.security.exception;

import java.util.Map;
import org.springframework.security.core.AuthenticationException;

/**
 * @author Peter Wu
 */
public class IllegalUserException extends AuthenticationException {

  private static final long serialVersionUID = 4634232939775284312L;

  private Map<String, String> errors;

  public IllegalUserException(String msg) {
    super(msg);
  }

  public IllegalUserException(String msg, Throwable t) {
    super(msg, t);
  }

  public Map<String, String> getErrors() {
    return errors;
  }

  public void setErrors(Map<String, String> errors) {
    this.errors = errors;
  }
}

File: security/src/main/kotlin/top/bettercode/summer/security/support/IllegalUserException.kt
package top.bettercode.summer.security.support

import org.springframework.http.HttpStatus
import org.springframework.security.authentication.InternalAuthenticationServiceException
import org.springframework.web.bind.annotation.ResponseStatus

/**
 * @author Peter Wu
 */
@ResponseStatus(HttpStatus.BAD_REQUEST)
class IllegalUserException : InternalAuthenticationServiceException {
    var errors: Map<String, String>? = null

    constructor(msg: String?) : super(msg)
    constructor(msg: String?, t: Throwable?) : super(msg, t)

}
